### PR TITLE
feat(store): warn about undecorated state class if Ivy is enabled in dev (both JIT/AOT)

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by arturovt** [comment]
Wait. Do not merge please.

**Comment by arturovt** [comment]
@markwhitfeld ping

File: packages/store/src/ivy/ensure-state-class-is-injectable.ts [hunk 1]
```diff
@@ -1,19 +1,21 @@
 import { StateClass } from '@ngxs/store/internals';
 
-import { ivyEnabledInJitMode } from './ivy-enabled-in-jit-mode';
+import { ivyEnabledInDevMode$ } from './ivy-enabled-in-dev-mode';
```
Comment (markwhitfeld) [inline packages/store/src/ivy/ensure-state-class-is-injectable.ts]: I don't think that the `$` on the end is necessary. That is an observable convention.

Comment (arturovt) [reply packages/store/src/ivy/ensure-state-class-is-injectable.ts]: OK. Tslint whines on the shadowed variable, what would you propose?

**Comment by arturovt** [comment]
Currently, unit tests will fail because micro-tasks are "fired&forgotten". I will add SSR tests for Ivy.

File: .lintstagedrc.json [hunk 2]
```diff
@@ -1,4 +1,3 @@
 {
-  "*.ts": ["prettier --write", "lint --fix", "git add"],
-  "*.{js,html,css,scss,md,json,yml}": ["prettier --write", "git add"]
+  "*.{ts,js,html,css,scss,md,json,yml}": ["prettier --write", "git add"]
```
Comment (splincode) [inline .lintstagedrc.json]: You can push current changes with another commit to master

**Comment by splincode** [comment]
@arturovt please merge into master

**Comment by arturovt** [comment]
> Todo(@arturovt): remove async behavior with `ReplaySubject`

Will do that this evening.

File: packages/store/src/ivy/ivy-enabled-in-dev-mode.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * Ivy doesn't expose these functions in production. Developers will be able
+ * to see warnings in both JIT/AOT modes, but only if an application
+ * is in development.
+ */
+export function setIvyEnabledInDevMode(): void {
+  try {
+    // `try-catch` will also handle server-side rendering, as
+    // `window is not defined` will not be thrown.
+    const ng = (window as any).ng;
+    const ivyEnabledInDevMode =
```
Comment (markwhitfeld) [inline packages/store/src/ivy/ivy-enabled-in-dev-mode.ts]: This shadows the function declared below with the same name. Make the names different.

Comment (markwhitfeld) [reply packages/store/src/ivy/ivy-enabled-in-dev-mode.ts]: See the linting error

File: .lintstagedrc.json [hunk 4]
```diff
@@ -1,4 +1,3 @@
 {
-  "*.ts": ["prettier --write", "lint --fix", "git add"],
```
Comment (markwhitfeld) [inline .lintstagedrc.json]: Why was the `lint --fix` removed?

Comment (arturovt) [reply .lintstagedrc.json]: Has 0 impact and fails in 90% of cases when running via `lint-staged`. `yarn lint --fix` returns 0 code and `lint --fix` (via lint-staged) throws with unknown reasons.

**Comment by splincode** [comment]
@arturovt 

image

**Comment by arturovt** [comment]
Sorry, not gonna debug Travis errors...

**Comment by splincode** [comment]
@arturovt maybe we can usage `cpx` in root package? instead children packages

**Comment by DanielHabenicht** [comment]
@markwhitfeld What warning should be displayed after this feature got merged? 
Just updated to angular 9 and all my states had no `@Injectable()` decorator, but no warning got displayed. Should I open a Issue?

**Comment by splincode** [comment]
@DanielHabenicht what your ngxs version? this works with 3.6.2+

**Comment by DanielHabenicht** [comment]
Yep it's 3.6.2 as I updated everything to the highest version. But I only got the `Can't resolve all parameters off state` Error.

**Comment by splincode** [comment]
create issue with github repo for reproduce

**Pull request merged**